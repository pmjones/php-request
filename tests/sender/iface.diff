001+ 
001- foo: bar
002- baz: zim
003+ Fatal error: Interface 'SapiResponseInterface' not found in /Users/pmjones/Code/pmjones/php-request/tests/MySapiResponse.inc on line 3
004- foo
005- int(400)
006- array(4) {
007-   [0]=>
008-   string(8) "foo: bar"
009-   [1]=>
010-   string(8) "baz: zim"
011-   [2]=>
012-   string(30) "Set-Cookie: cookie1=v1%26%25v2"
013-   [3]=>
014-   string(38) "Content-type: text/html; charset=UTF-8"
015- }