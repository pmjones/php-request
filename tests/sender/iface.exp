foo: bar
baz: zim
--HEADERS--
foo
int(400)
array(4) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(8) "baz: zim"
  [2]=>
  string(30) "Set-Cookie: cookie1=v1%26%25v2"
  [3]=>
  string(38) "Content-type: text/html; charset=UTF-8"
}