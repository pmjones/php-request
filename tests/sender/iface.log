
---- EXPECTED OUTPUT
foo: bar
baz: zim
--HEADERS--
foo
int(400)
array(4) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(8) "baz: zim"
  [2]=>
  string(30) "Set-Cookie: cookie1=v1%26%25v2"
  [3]=>
  string(38) "Content-type: text/html; charset=UTF-8"
}
---- ACTUAL OUTPUT

--HEADERS--
Fatal error: Interface 'SapiResponseInterface' not found in /Users/pmjones/Code/pmjones/php-request/tests/MySapiResponse.inc on line 3
---- FAILED
