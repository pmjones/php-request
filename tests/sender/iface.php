<?php
include "../MySapiResponse.inc";
$response = new MySapiResponse();
$response->setCode(400);
$response->setContent('foo');
$response->setHeader('Foo', 'bar');
$response->setCookie('cookie1', 'v1&%v2');
$response->addHeaderCallback(function($response) {
    $response->addHeader('Baz', 'zim');
});
(new SapiResponseSender())->send($response);
echo "\n";
var_dump(http_response_code());
var_dump(headers_list());
