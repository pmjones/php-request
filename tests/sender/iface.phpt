--TEST--
SapiResponseSender::send with custom SapiResponseInterface
--CGI--
--INI--
expose_php=0
--FILE--
<?php
include "../MySapiResponse.inc";
$response = new MySapiResponse();
$response->setCode(400);
$response->setContent('foo');
$response->setHeader('Foo', 'bar');
$response->setCookie('cookie1', 'v1&%v2');
$response->addHeaderCallback(function($response) {
    $response->addHeader('Baz', 'zim');
});
(new SapiResponseSender())->send($response);
echo "\n";
var_dump(http_response_code());
var_dump(headers_list());
--EXPECTHEADERS--
foo: bar
baz: zim
--EXPECT--
foo
int(400)
array(4) {
  [0]=>
  string(8) "foo: bar"
  [1]=>
  string(8) "baz: zim"
  [2]=>
  string(30) "Set-Cookie: cookie1=v1%26%25v2"
  [3]=>
  string(38) "Content-type: text/html; charset=UTF-8"
}
