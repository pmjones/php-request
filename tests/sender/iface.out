
--HEADERS--
Fatal error: Interface 'SapiResponseInterface' not found in /Users/pmjones/Code/pmjones/php-request/tests/MySapiResponse.inc on line 3